frontend/components/keypad-dialog.js

```diff
@@ -96,98 +109,52 @@ Tonic.add(class KeypadDialog extends Tonic {
   }
 })
 
-// Function to remove the last char in string
 function RemoveLastChar (value) {
   value = value.slice(0, -1)
   return value
 }
 
-export async function Keypad () {
-  console.log('KeyPad Triggered')
-}
-
-// async function promptPINold (Incorrect) {
-//   const [$secret, setSecret] = write('')
-//   document.getElementById('KeyPadDisplayUnlock').placeholder = ''
-
-//   if (Incorrect) {
-//     document.getElementById('KeyPadDisplayUnlock').value = ''
-//     document.getElementById('KeyPadDisplayUnlock').placeholder = 'Incorrect PIN'
-//   }
-
-//   nEl('edit-keypad-dlg2').open = true
-//   nClick('unlock-button', async () => {
-//     const value = document.getElementById('KeyPadDisplayUnlock').value
-//     setSecret(value)
-//     document.getElementById('KeyPadDisplayUnlock').value = ''
-//     nEl('edit-keypad-dlg2').open = false
-//   })
-//   const returnValue = await next($secret)
-//   console.log('[keypad-dialog] returnValue:', returnValue)
-//   return returnValue
-// }
-
-function showKeypad () {
-  nEl('edit-keypad-dlg2').open = true
-}
-function hideKeypad () {
-  document.getElementById('KeyPadDisplayUnlock').placeholder = ''
-  document.getElementById('KeyPadDisplayUnlock').value = ''
-  nEl('edit-keypad-dlg2').open = false
+async function psuhSecret (secret) {
+  await kernel.setSecret(secret)
+  await kernel.encryptMessage(secret)
 }
 
-async function tryDecrypt (message, secret) {
-  return new Promise((resolve, reject) => {
-    setTimeout(() => {
-      resolve(decrypt(message, secret) !== '')
-    }, 1000)
-  })
-
-  // const unlocked = (await decrypt(message, secret) !== '')
-  // return unlocked
+async function tryUnlock () {
+  const secret = get($secret)
+  const text = get($text)
+  try {
+    const unlock = await decrypt(text, secret)
+    if (unlock !== '') {
+      setUnlockState(true)
+      setText(unlock)
+    } else {
+      document.getElementById('KeyPadDisplayUnlock').placeholder = 'Incorrect PIN'
+    }
+  } catch (e) {
+    if (e.message === 'Malformed UTF-8 data') {
+      document.getElementById('KeyPadDisplayUnlock').placeholder = 'Incorrect PIN?'
+      console.info('Actually an error occured. If this happend while entering the correct PIN try reloading the page.')
+    }
+  }
 }
 
 async function waitForInput (text) {
-  const x = await next($secret)
-  const unlock = await decrypt(text, x)
-  console.log('UNLOCK: ', unlock)
-  if (unlock !== '') { return x } else {
-    waitForInput(text)
-  }
+  const state = await next($unlockState)
+  if (state) {
+    return get($text)
+  } else return waitForInput()
 }
 
 export async function promptPIN (text) {
```

Comment:
An awaitable function that waits for user to input secret PIN before decrypting and returning the message to render.